File: fibonacci.nim

import times
import strutils

proc fibonacci*(n: int64): int64 =
  if n <= 1'i64:
    1'i64
  else:
    fibonacci(n - 1'i64) + fibonacci(n - 2'i64)

proc main() =
  let starttime = times.epochTime()
  echo "fibonacci(46) = ", fibonacci(46)
  let endtime = times.epochTime()
  echo "elapsed: $# sec" % [$(endtime - starttime)]

main()

File: fibonacci.rs

use std::time::SystemTime;

fn main() {
    let start = SystemTime::now();
    println!("fibonacci(46) = {}", fibonacci(46));
    match start.elapsed() {
        Ok(elapsed) => {
            println!("elapsed: {}.{} sec", elapsed.as_secs(), elapsed.subsec_nanos());
        }
        Err(e) => {
            println!("Error: {:?}", e);
        }
    }
}

fn fibonacci(n: i64) -> i64 {
    if n <= 1 {
        1
    } else {
        fibonacci(n - 1) + fibonacci(n - 2)
    }
}
